Sources/AuthFoundation/Token Management/Internal/KeychainTokenStorage.swift

```diff
@@ -87,69 +96,74 @@ class KeychainTokenStorage: TokenStorage {
         }
     }
     
-    func replace(token id: String, with token: Token) throws {
-        let oldResult = try Keychain
+    func replace(token id: String, with token: Token, security: [Credential.Security]?) throws {
+        guard let oldResult = try Keychain
             .Search(account: id,
                     service: KeychainTokenStorage.serviceName)
-            .get()
-        
-        let oldToken = try self.token(with: oldResult)
-        try oldResult.delete()
+                .list()
+                .first
+        else {
+            throw TokenError.cannotReplaceToken
+        }
         
         token.id = id
         
         let data = try encoder.encode(token)
-        try Keychain
-            .Item(account: id,
-                  service: KeychainTokenStorage.serviceName,
-                  accessibility: .unlocked,
-                  value: data)
-            .save()
+        let accessibility = security?.accessibility ?? .afterFirstUnlock
```

Comment:
Should we derive this setting from `oldResult`?

Reply:
The user could potentially change the accessibility settings, but you're correct in that it should default to the current value.